File: src/no/omb/bigboat/data/ClubEntry.java
package no.omb.bigboat.data;

import java.text.DecimalFormat;

import no.omb.bigboat.BigBoat;

public class ClubEntry implements Comparable<ClubEntry> {
	
	private String club;
	private double score;
	private int numScores = 0;

	public ClubEntry(String club) {
		this.club = club;
	}

	public String getClub() {
		return club;
	}

	public double getScore() {
		return score;
	}

	public void setScore(double score) {
		this.score = score;
	}

	public int getNumScores() {
		return numScores;
	}

	public void setNumScores(int numScores) {
		this.numScores = numScores;
	}

	@Override
	public String toString() {
		DecimalFormat df = new DecimalFormat("#0.00");
		return club + BigBoat.SEP + df.format(score/BigBoat.CLUB_MAX_SCORES);
	}

	@Override
	public int compareTo(ClubEntry o) {
		return (int) (score*100 - o.score*100);
	}

}

File: src/no/omb/bigboat/data/ClubEntry.kt
package no.omb.bigboat.data

import no.omb.bigboat.BigBoat
import java.text.DecimalFormat

data class ClubEntry(
    val club: String,
    var score: Double = 0.0,
    var numScores: Int = 0
) : Comparable<ClubEntry> {

    override fun toString(): String {
        val df = DecimalFormat("#0.00")
        return club + BigBoat.SEP + df.format(score / BigBoat.CLUB_MAX_SCORES)
    }

    override fun compareTo(other: ClubEntry): Int {
        return (score * 100 - other.score * 100).toInt()
    }
}
